pkg/controller/certificates/trigger/policies/policies_test.go

```diff
@@ -208,8 +212,9 @@ func TestDefaultPolicyChain(t *testing.T) {
 			message: "Issuing certificate as Secret was previously issued by IssuerKind.new.example.com/testissuer",
 			reissue: true,
 		},
-		// we only have a basic test here for this as unit tests for the
-		// `certificates.RequestMatchesSpec` function cover all other cases.
+		// We only have a basic test here about what should be triggering a
```

Comment:
I wonder if we should actually have a unit test for that [`RequestMatchesSpec`](https://github.com/jetstack/cert-manager/blob/master/pkg/controller/certificates/util.go#L118) function

Reply:
Yes, there should be a unit test on `RequestMatchesSpec`. At first, I wanted to add a unit test for  `RequestMatchesSpec`, I quickly realized that I didn't want to "just" test `RequestMatchesSpec`: I wanted to test the entire policy chain used by the trigger controller.

I'll add a unit test for `RequestMatchesSpec`.

Reply:
> I wanted to test the entire policy chain used by the trigger controller.
That makes sense.

However, as there already is one test in the same file that also tests this policy chain, I think it would be better to merge the two as opposed to having two different unit tests with different syntaxes for the same function

Reply:
I moved 90% of the test cases to the new syntax, I still need to move the 4 last:

- trigger renewal if renewalTime is right now
- trigger renewal if renewalTime is in the past
- does not trigger renewal if the x509 cert has been re-issued, but Certificate's renewal time has not been updated yet
- does not trigger renewal if renewal time is in 1 minute

I'll admit, they do not fit into the new syntax but I'll try to merge them too tomorrow morning.